house-cats/huyan
# Initialize backend code to api/ directory

*(This addresses part of #68.)*

Following #67, the plan's to have this repo broken out into three npm packages in the following directories:

- `api/` backend code
    - This is what this PR does!
- `lib/` library code for frontend/backend sharing
    - Not present yet; will do in a separate PR that bootstraps the library.
- `web/` frontend code
    - Done in #67.

In addition, this sets up some more infrastructure for the backend to get things going:

- __Testing with Jest:__
    - `npm install --save-dev jest`
    - Set up `npm test` target to run `jest`.
    - Add `hello.test.js` as a basic test to make sure that `jest`'s infrastructure works.
    - Run `npm test` to make sure that tests pass.
- __Formatting with Prettier:__
    - Use the same `.prettierrc` as we've used in `web/` for consistent formatting.
- __Precommit Hooks with Husky and Lint Staged:__
    - `npm install --save-dev husky lint-staged`
    - Follow steps from #69, running `npm husky-init` and then adjusting `package.json` to account for `api/` being a subdirectory in the repo instead of top-level lateral to the `.git/` directory.
- __Dependency Upgrades with Dependabot:__ 
    - Add separate entry in `dependabot.yml` for the `api/` directory in the same vein as `web/` currently has it.
- __CI/CD with GitHub Actions:__
    - Rename `.github/workflows/web.yml` to `.github/workflows/build.yml` and set up separate jobs for frontend and backend.
